File: app/src/main/java/com/sk/revisit/helper/NetHelper.java
package com.sk.revisit.helper;

import android.util.Log;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class NetHelper {
    final String TAG = NetHelper.class.getSimpleName();
    final OkHttpClient okHttpClient;

    public NetHelper(OkHttpClient client) {
        this.okHttpClient = client;
    }

    public Response head(String url) {
        Request request = new Request.Builder().head().url(url).build();
        long size;
        try {
            okHttpClient.newCall(request).enqueue(new Callback() {
                @Override
                public void onFailure(@NonNull Call call, @NonNull IOException e) {

                }

                @Override
                public void onResponse(@NonNull Call call, @NonNull Response response) {
                    assert response.body() != null;
                    // size = response.body().contentLength();
                }
            });
        } catch (Exception e) {
            Log.e(TAG, e.toString());
        }
        return null;
    }
}

File: app/src/main/java/com/sk/revisit/helper/NetHelper.kt
package com.sk.revisit.helper

import android.util.Log
import okhttp3.*
import java.io.IOException

class NetHelper(private val okHttpClient: OkHttpClient) {
    private val TAG = NetHelper::class.java.simpleName

    fun head(url: String): Response? {
        val request = Request.Builder().head().url(url).build()
        try {
            okHttpClient.newCall(request).enqueue(object : Callback {
                override fun onFailure(call: Call, e: IOException) {
                    Log.e(TAG, "Head request failed", e)
                }

                override fun onResponse(call: Call, response: Response) {
                    response.body?.let {
                        // size = it.contentLength()
                    }
                    response.close()
                }
            })
        } catch (e: Exception) {
            Log.e(TAG, e.toString())
        }
        return null
    }
}
